### ISSUE TITLE
WASM -> TypeError: Cannot read property 'cwrap' of undefined

### ISSUE THREAD

**Issue by sbstcano** [tensorflow/tfjs#3864]
Hi,
I'm trying to use wasm with facemesh but keep getting this when trying to set wasm as backend.
I'm using version 2.3.0. And it's an Angular 9 project.

```
import * as tf from '@tensorflow/tfjs-core'
import * as facemesh from '@tensorflow-models/facemesh'
import {setWasmPath} from "@tensorflow/tfjs-backend-wasm"

setWasmPath(`https://cdn.jsdelivr.net/npm/@tensorflow/tfjs-backend-wasm@2.3.0/dist/tfjs-backend-wasm.wasm`)

tf.setBackend('wasm')
```

```
TypeError: Cannot read property 'cwrap' of undefined
    at Object.setupFunc (tf-backend-wasm.esm.js:17)
    at engine.js:161
    at Array.forEach (<anonymous>)
    at Engine.setupRegisteredKernels (engine.js:159)
    at Engine.setBackend (engine.js:152)
    at Module.setBackend (globals.js:258)
    at Module../src/app/enrollment/enrollment.component.ts (enrollment.component.ts:20)
    at __webpack_require__ (bootstrap:79)
    at Module../src/app/app-routing.module.ts (main.js:450)
    at __webpack_require__ (bootstrap:79)
```

**Comment by deel77** [comment]
I think I have identified main problem. Angular is overwriting `Promise.resolve()` function with its own wrapping function in order to use `ZoneAwarePromise` as a return type in order to automatically trigger change detection after promise finishes.

And TF-Core is using following code to test if created backend is Promise (which is Promise type only for WASM backend, as to my knowledge):
```
// Test if the factory returns a promise.
if (Promise.resolve(backend) === backend) {
```
see https://github.com/tensorflow/tfjs/blob/master/tfjs-core/src/engine.ts#L294

and this type of test does not return true as return type of `Promise.resolve(backend)` is not same `Promise` but `ZoneAwarePromise` wrapper of original `Promise`.

I have tried to run `tf.setBackend('wasm')` inside `ngZone.runOutsideAngular()` but it does not help. I would try something more.
